videoproviders/api/videofront.py

```diff
@@ -168,6 +169,35 @@ def get_video_with_subtitles(self, video_id):
         """Return a single video, identified by its id, along with the
         subtitles information. This method is used by the video xblock to fetch
         video + subtitles info in one go."""
+        if getattr(settings, "VIDEOFRONT_CDN_BASE_URL"):
+            # Try getting the video document directly from the S3 bucket metadata file
+            try:
+                response = requests.get(
+                    "{cdn:s}/videos/{id:s}/metadata.json".format(
+                        cdn=settings.VIDEOFRONT_CDN_BASE_URL, id=video_id
+                    )
+                )
+            except requests.exceptions.RequestException:
+                logger.error(
+                    'Unexpected error loading metadata file for video "%s".',
+                    video_id,
+                )
+            else:
+                if response.status_code == 200:
+                    try:
+                        return response.json()
+                    except JSONDecodeError:
+                        logger.error(
+                            'Metadata for VideoFront video "%s" is not valid json.',
+                            video_id,
+                        )
+                else:
+                    logger.error(
+                        'VideoFront video "%s" could not be loaded from metadata file.',
+                        video_id,
+                    )
```

Comment:
Nice! :+1: 

Just a suggestion to avoid multiple try blocks:

```suggestion
        if getattr(settings, "VIDEOFRONT_CDN_BASE_URL"):
            # Try getting the video document directly from the S3 bucket metadata file
            try:
                response = requests.get(
                    "{cdn:s}/videos/{id:s}/metadata.json".format(
                        cdn=settings.VIDEOFRONT_CDN_BASE_URL, id=video_id
                    )
                )
                if response.status_code == 200:
                    return response.json()
                logger.error(
                    'VideoFront video "%s" could not be loaded from metadata file.',
                    video_id,
                )
            except requests.exceptions.RequestException:
                logger.error(
                    'Unexpected error loading metadata file for video "%s".',
                    video_id,
                )
            except JSONDecodeError:
                logger.error(
                    'Metadata for VideoFront video "%s" is not valid json.',
                    video_id,
                )

```

Reply:
won't take as I prefer to try/catch one by one.